### DISCUSSION TITLE
Is it possible to vertically center NF symbols?

### DISCUSSION THREAD

**Discussion by mbenford** [kovidgoyal/kitty#6682] [Q&A] [answered]
I've recently started to use more NF symbols on my configurations and noticed that many of them (maybe all?) are taller than regular characters, and appear misaligned when next to numbers and letters. Here are some examples:

![image](https://github.com/kovidgoyal/kitty/assets/446867/1daf7ffb-120a-416c-bd80-13ac74ba64a6)
![image](https://github.com/kovidgoyal/kitty/assets/446867/d86eef14-a58e-4efb-86fd-c156e429a9fb)

After reading the docs I think `modify_font baseline X` is what I'm looking for, but it's my understanding that it applies to all characters, right? Is it possible to specify what code points should be affected, as `narrow_symbols` does?

Also, I'm probably wrong, but I recall Kitty rendering those symbols centered in an older version (can't remember which one, though). Is my memory failing me?

**Comment by kovidgoyal** [comment]
There is no per glyph adjustment in kitty. I suggest you use nerd fonts directly not via  patched font. And kitty had no special rendering for them, ever.

**Reply by mbenford** [reply]
Thanks! Yeah, I'm aware of the issues with patched fonts and don't use them.

Do you think it would make sense for `modify_font` to support individual code points? I don't know if I'm skilled enough to work on that, but I could give it a shot.

**Comment by kovidgoyal** [comment]
On Sat, Oct 07, 2023 at 09:18:20AM -0700, Michael Benford wrote:
> Thanks! Yeah, I'm aware of the issues with patched fonts and don't use them.
> 
> Do you think it would make sense for `modify_font` to support individual code points? I don't know if I'm skilled enough to work on that, but I could give it a shot.

Not really, modify_font is font wide. You would need a separate
modify_glyph setting, but really why not just fix the baseline for those
symbols in their font files?

**Reply by mbenford** [reply]
I see your point. Thanks again.

**Comment by mbenford** [comment, marked as answer]
Ok, so I followed @kovidgoyal's advice and tried to use a NF font with an appropriate baseline, but I couldn't find a version of the standalone `Symbols Nerd Font` that worked. So I decided to use the patched version of my current font - which has a better baseline for symbols - and manually map all code points by using `symbol_map`. To my surprise, I didn't need to do that because Kitty automagically picked up the patched font only for symbols! I was aware that it would try to use the `Symbols Nerd Font` if available, but not the patched version of the regular font in use. At first I thought Kitty was using the patched font to render all characters, but `--debug-font-fallback` confirmed that it was using it only for symbols.

So, problem solved by Kitty itself! :D
